### ISSUE TITLE
Could you provide data explanation for the releasing data？

### ISSUE THREAD

**Issue by xuyifan-0731** [DigiRL-agent/digirl#1]

**Comment by xuyifan-0731** [comment]
In pre-collected trajectories， I found data like this:


```
Element 1450:
[{'observation': "Previous Actions: Goal: What's a good restaurant in Los Angeles?</s>", 'next_observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.7852, 0.6779]", "lift_point": "[0.7852, 0.6779]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'image_features': array([-0.4997673 ,  2.460607  ,  0.05416491, ...,  0.38946685,
        1.8913703 , -0.43919575], dtype=float32), 'image_path': '/home/yifeizhou/logs/zeroshot-collect-general-sendkey-temp1.0-2/images/test3/1715155212.891637_1.png', 'next_image_features': array([-1.4063659,  1.9205499,  0.272918 , ...,  0.8451029,  0.7577356,
       -1.0199058], dtype=float32), 'task': "What's a good restaurant in Los Angeles?", 'reward': 0, 'done': False, 'action': 'Action Plan: [DUAL_POINT,DUAL_POINT,PRESS_HOME,DUAL_POINT] ; Action Decision: "action_type": "DUAL_POINT", "touch_point": "[0.7852, 0.6779]", "lift_point": "[0.7852, 0.6779]", "typed_text": ""', 'trajectory_reward': 1, 'mc_return': 0.8145062499999999}, {'observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.7852, 0.6779]", "lift_point": "[0.7852, 0.6779]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'next_observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.3315, 0.4387]", "lift_point": "[0.3315, 0.4387]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'image_features': array([-1.4063659,  1.9205499,  0.272918 , ...,  0.8451029,  0.7577356,
       -1.0199058], dtype=float32), 'image_path': '/home/yifeizhou/logs/zeroshot-collect-general-sendkey-temp1.0-2/images/test3/1715155212.891637_2.png', 'next_image_features': array([-0.13373429,  2.9066284 ,  0.60418534, ...,  1.9622939 ,
        1.3106302 , -0.43112513], dtype=float32), 'task': "What's a good restaurant in Los Angeles?", 'reward': 0, 'done': False, 'action': 'Action Plan: [DUAL_POINT,TYPE,DUAL_POINT,TYPE] ; Action Decision: "action_type": "DUAL_POINT", "touch_point": "[0.3315, 0.4387]", "lift_point": "[0.3315, 0.4387]", "typed_text": ""', 'trajectory_reward': 1, 'mc_return': 0.8573749999999999}, {'observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.3315, 0.4387]", "lift_point": "[0.3315, 0.4387]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'next_observation': 'Previous Actions: "action_type": "TYPE", "touch_point": "[-1.0, -1.0]", "lift_point": "[-1.0, -1.0]", "typed_text": "Whats a good restaurant in Los Angeles?" Goal: What\'s a good restaurant in Los Angeles?</s>', 'image_features': array([-0.13373429,  2.9066284 ,  0.60418534, ...,  1.9622939 ,
        1.3106302 , -0.43112513], dtype=float32), 'image_path': '/home/yifeizhou/logs/zeroshot-collect-general-sendkey-temp1.0-2/images/test3/1715155212.891637_3.png', 'next_image_features': array([-1.084147  ,  2.7585857 ,  0.30194184, ...,  0.18133342,
        0.34699765,  1.1443771 ], dtype=float32), 'task': "What's a good restaurant in Los Angeles?", 'reward': 0, 'done': False, 'action': 'Action Plan: [TYPE,DUAL_POINT,DUAL_POINT,DUAL_POINT] ; Action Decision: "action_type": "TYPE", "touch_point": "[-1.0, -1.0]", "lift_point": "[-1.0, -1.0]", "typed_text": "Whats a good restaurant in Los Angeles?"', 'trajectory_reward': 1, 'mc_return': 0.9025}, {'observation': 'Previous Actions: "action_type": "TYPE", "touch_point": "[-1.0, -1.0]", "lift_point": "[-1.0, -1.0]", "typed_text": "Whats a good restaurant in Los Angeles?" Goal: What\'s a good restaurant in Los Angeles?</s>', 'next_observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.1250, 0.6632]", "lift_point": "[0.1250, 0.6632]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'image_features': array([-1.084147  ,  2.7585857 ,  0.30194184, ...,  0.18133342,
        0.34699765,  1.1443771 ], dtype=float32), 'image_path': '/home/yifeizhou/logs/zeroshot-collect-general-sendkey-temp1.0-2/images/test3/1715155212.891637_4.png', 'next_image_features': array([-0.73024786,  2.5167723 , -0.32905063, ...,  1.3304031 ,
        0.37446848, -0.67203707], dtype=float32), 'task': "What's a good restaurant in Los Angeles?", 'reward': 0, 'done': False, 'action': 'Action Plan: [DUAL_POINT,STATUS_TASK_COMPLETE] ; Action Decision: "action_type": "DUAL_POINT", "touch_point": "[0.125, 0.6632]", "lift_point": "[0.125, 0.6632]", "typed_text": ""', 'trajectory_reward': 1, 'mc_return': 0.95}, {'observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.1250, 0.6632]", "lift_point": "[0.1250, 0.6632]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'next_observation': 'Previous Actions: "action_type": "DUAL_POINT", "touch_point": "[0.7604, 0.9249]", "lift_point": "[0.7604, 0.9249]", "typed_text": "" Goal: What\'s a good restaurant in Los Angeles?</s>', 'image_features': array([-0.73024786,  2.5167723 , -0.32905063, ...,  1.3304031 ,
        0.37446848, -0.67203707], dtype=float32), 'image_path': '/home/yifeizhou/logs/zeroshot-collect-general-sendkey-temp1.0-2/images/test3/1715155212.891637_5.png', 'next_image_features': array([-1.5112836 ,  3.0659556 , -0.19630237, ..., -1.0965215 ,
        0.12311126,  1.1756645 ], dtype=float32), 'task': "What's a good restaurant in Los Angeles?", 'reward': 1, 'done': True, 'action': 'Action Plan: [DUAL_POINT,DUAL_POINT,DUAL_POINT,TYPE] ; Action Decision: "action_type": "DUAL_POINT", "touch_point": "[0.7604, 0.9249]", "lift_point": "[0.7604, 0.9249]", "typed_text": ""', 'trajectory_reward': 1, 'mc_return': 1.0}]
```
        
 Could you provide an introduction of what reward,trajectory_reward, and mc_return mean and how to build these reward according to your paper？ It seems that no value model has been released in your repo.
 
 Thanks a lot.

**Comment by BiEchi** [comment]
Hi, each rollout has the format below.

<img width="842" alt="image" src="https://github.com/DigiRL-agent/digirl/assets/60613238/2345b0e9-f26d-46c7-b938-61c6bed058d3">

`reward`: transition reward.
`trajectory_reward`: our reward is sparse. If the reward of the final step is 1 (which means that the agent successfully completes the task inside `max_steps`), the `trajectory_reward` of all steps will become 1.
`mc_return`: just monte-carlo return. In the paper, it's represented as $\lambda^{H-h}r(s_H, a_H, c)$ as shown in Equation (4.3).
Value model: This is filtered AWR, so the AutoUI agent can infer without the value model. If you want to train the checkpoint yourself, you will automatic obtain the value model when saving a checkpoint.

Sidenote: `image_path` in each rollout corresponds to the screenshot *after* the transition.

**Comment by xuyifan-0731** [comment]
Thank you very much for your patience in explaining! This is very helpful to me

**Comment by zhengshf** [comment]
The trajectory_reward of all steps in the trajectory is the same. Why is the same trajectory_reward stored in each step? Thanks alot.@BiEchi

**Comment by zhengshf** [comment]
I saw rollout.mean\max\min in the code, but it was not used. Are these parameters intended to be used in the future to select a training set with appropriate task values?
